lib/lin_api.ml

```diff
@@ -179,18 +184,48 @@ module MakeCmd (ApiSpec : ApiSpec) : Lin.CmdSpec = struct
   let gen_printer : type a r. string -> (a,r,t) Fun.fn -> (a,r) Args.args -> string = fun name fdesc args ->
     name ^ " " ^ show_args fdesc args
 
+  (* Extracts a QCheck shrinker for argument lists *)
+  let rec gen_iter_of_desc
```

Comment:
Good point.
I first wrote the function in an "expanded version" using `((a, r) Args.args) -> ((a, r) Args.args) QCheck.Iter.t`
instead of the shorter `((a, r) Args.args) QCheck.Shrink.t`.
After getting it working I cleaned it up a bit - but clearly not enough :sweat_smile: 

I'll rename to `gen_shrinker_of_desc` as I agree - it is a better name!